blog.md

```diff
@@ -0,0 +1,203 @@
+
+When calculating each block, we will only need to load matrices with size $BM \times BK$ and $BK \times BN$ for~$K / BK$ times from global memory to shared memory.
+In that case, the total data loading will be reduce to 
+$$
+    M \times N \times K \times \left( \frac{1}{BM} + \frac{1}{BN} \right)
+$$
+which is much smaller than the naive way.
+
+Then in each block, we further tile the matrix and use the registers to store the data, as shown by
+
+![Fig.1](figs/thread.png)
+
+The tiled target matrix will be further divided as small matrices with size $TM \times TN$, and each thread will be used to calculate one of the tiled matrix.
+During this process, the outer product way is used and data will be loaded from shared memory to registers, with the amount of $(TM + TN) \times BK$ in total.
+
+Then we replaced the operations directly.
+As we mentioned above, the `Tropical` type is `bitstype`, and the data storaged in memeories are simply the floating point numbers, which can be directly used by CUDA kernels.
+Although the tropical algebra can not use the fused multiple add core (FMA), we found that the operation add/mul and max/min can be done on FMA and ALU parallelly, which means that we can use the FMA to do the add/mul and ALU to do the max/min at the same time.
+Then after all the calculation is done, the target in the registers will be stored back to global memory directly.
+
+For the boundary elements, a padding strategy is used, we simply set the element which are not acctually in the matrix as the zero element of the corresponding algebra, so that they will not effect the result of the calculation.
+In our package, we set the parameters as
+$$
+    BM = 64,~BK = 32,~BN = 64,~TM = TN = 4.
+$$
+
+In the end, we wrapped the code in Julia, and overloaded the function `LinearAlgebra.mutmul!`, so that a simple $*$ for tropical matrices will call our function and use GPU in calculations.
+Here is an example:
+```julia
+julia> using CUDA, LinearAlgebra, TropicalNumbers, CuTropicalGEMM
+
+julia> A = CuArray(Tropical.(rand(2,2)))
+2×2 CuArray{Tropical{Float64}, 2, CUDA.Mem.DeviceBuffer}:
+  0.5054551076120295ₜ  0.2566654342554737ₜ
+ 0.40277483290611305ₜ  0.8717314798683612ₜ
+
+julia> B = CuArray(Tropical.(rand(2,2)))
+2×2 CuArray{Tropical{Float64}, 2, CUDA.Mem.DeviceBuffer}:
+ 0.7488281325136905ₜ  0.03728702805013795ₜ
+ 0.8437060742174199ₜ   0.9777629175478465ₜ
+
+julia> A * B
+2×2 CuArray{Tropical{Float64}, 2, CUDA.Mem.DeviceBuffer}:
+   1.25428324012572ₜ  1.2344283518033201ₜ
+ 1.7154375540857811ₜ  1.8494943974162077ₜ
+```
+It is much faster than the previous CPU implementation `TropicalGEMM.jl`
+```julia
+julia> A = Tropical.(rand(1024,1024));
+
+julia> B = Tropical.(rand(1024,1024));
+
+julia> @time A * B;
+  4.215667 seconds (6 allocations: 8.030 MiB)
+
+julia> using TropicalGEMM
+
+julia> @time A * B;
+  0.095628 seconds (2 allocations: 8.000 MiB)
+
+julia> CuA = CuArray(A);
+
+julia> CuB = CuArray(B);
+
+julia> @time CuA * CuB;
```

Comment:
1. Please emphasis a second run.
2. It is not enough for compare wtih CPU, please also compare with the fall back implementation with map-reduce.
3. I remember that you have removed the synchronize operation? Why can you use `@time` here?

Reply:
That's a mistake, I removed that part, and showed an example in the 'how to use' part